dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -169,7 +169,9 @@ def login
       auth_hash = inject_clever_data(auth_hash)
     end
 
-    user = User.from_omniauth(auth_hash, auth_params, request)
+    params = auth_params.presence || {}
+    params[:user_type] = cookies['sign_up_user_type'] unless params[:user_type]
+    user = User.from_omniauth(auth_hash, params, request)
```

Comment:
Setting the `:user_type` from the cookie value in the `auth_params` if it is not present.